packages/lib/src/item-input/object.ts

```diff
@@ -218,7 +214,7 @@ export function objectDef<T extends object>(name: string, inputDefsByProperty: I
 								const laterPropertyInputDefinition = inputDefsByProperty[propertyName]
 								const updateIfNeeded = laterPropertyInputDefinition.updateIfNeeded
 								if (updateIfNeeded) {
-									const laterPropertyValue = await updateIfNeeded(updated[propertyName], updatedPropertyName, [{ ...updated }, ...context])
+									const laterPropertyValue = await updateIfNeeded(updated[propertyName], action, [{ ...updated }, ...context])
```

Comment:
Another thing I noticed while working on the unit tests. The `updatedPropertyName` is fine for top-level properties (line 105) but for nested properties, it only tells us which object the nested property is in and not property name inside that object that is actually being edited.

A real-world example of this is the `clientName` definition on line 58 of the `apps:create` command (`packages/cli/src/commands/apps/create.ts`). Before this fix, it would pass just `oauth` but after this it will pass `oauth.clientName`.

(There is currently no code that uses this so the difference doesn't matter ATM but we should be passing on everything.)